src/catalua.h

```diff
@@ -3,9 +3,67 @@
 #define CATALUA_H
 
 #include "int_id.h"
+#include "enums.h"
+#include "item.h"
 
 #include <string>
 #include <sstream>
+#include <vector>
+#include <list>
+
+#ifdef LUA
+extern "C" {
+#include "lua.h"
+#include "lualib.h"
+#include "lauxlib.h"
+}
+#endif //LUA
+
+enum CallbackArgumentType : int {
+    Undefined = -1,
+    Integer = 0,
+    Number,
+    Double = Number,
+    Float = Number,
+    String,
+    Tripoint,
+    Item
+};
+
+struct CallbackArgument {
+    std::string name = "";
+    CallbackArgumentType type = CallbackArgumentType::Undefined;
```

Comment:
All constructors set this member explicitly, the default value defined here is never used. You don't need it and why should a `CallbacArgument` ever have a undefined type (you don't need that enumeration value as well)?